modules/generic_worker/templates/run-generic-worker.py.erb

```diff
@@ -0,0 +1,101 @@
+#!/usr/local/bin/python3
+
+import sys
+import imp
```

Comment:
We can use the following approach: 
1. Run generic-worker from shell script (like now)
2. use python to treat generic-worker exit code and open/close bugs

I really want to keep run-generic-worker shell scrip simple, something like this:

rm -f tasks-resolved-count.txt

export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin
source /usr/local/share/generic-worker/bugzilla-utils.sh

# First run the generic-worker, passing through any arguments handed to this
# wrapper script...
/usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
exitstatus="${PIPESTATUS[0]}"

python3 generic-worker-utils.py $exitstatus

Dave, are you OK with this approach?

Reply:
+1
I think this is more robust because it will still work if we have python issues (or other problems in the exit handling like with bugzilla), and we won't have the python interpreter as a parent for the worker (may not be a security or performance issue, but it feels like it to me ;).

To me, this reads like "always run generic-worker, and do something based on the exit code". Does that match the purpose?

You could also remove the sourcing of the bugzilla-utils.sh, and the rm of tasks-resolved-count.txt (that is only used by the python script, right?).